server/src/main/java/io/crate/execution/dml/Indexer.java

```diff
@@ -747,38 +801,74 @@ public ParsedDocument index(IndexItem item) throws IOException {
         );
         Object[] values = item.insertValues();
 
-        for (int i = 0; i < values.length; i++) {
-            Reference reference = columns.get(i);
-            Object value = valueForInsert(reference.valueType(), values[i]);
-            ColumnConstraint check = columnConstraints.get(reference.column());
-            if (check != null) {
-                check.verify(value);
-            }
-            if (reference.granularity() == RowGranularity.PARTITION) {
-                continue;
-            }
-            ValueIndexer<Object> valueIndexer = (ValueIndexer<Object>) valueIndexers.get(i);
-            if (value == null) {
-                continue;
+        if (indexOrder.isEmpty()) {
```

Comment:
would imho be good if we could unify the indexOrder empty/nonEmpty case. it is doing the same except for determining the iteration order.

E.g. either extract the body in a function, or always create an indexOrder

Reply:
Thank you, re-adding indexOrder for update/insert.